### PR TITLE
feat: execute python script in distributed mode

### LINKED ISSUE #411: System table in distribute mode
## Background
In standalone mode, datanode creates system table and regions during it's first startup, and registers these tables to catalog.
But in distributed mode, all regions of system tables should be treated equally. For example , for system table `scripts`, when frontend received a `insert into scripts` request, it should forward the create request to all datanodes, instead of forwarding by partition rules.

## Proposal
Frontend should be able to distinguish system table request. System table names should be retrived from metasrv.

For insert request, system table insert requests should bypass current writer splitter logic and forward to all datanodes.
https://github.com/GreptimeTeam/greptimedb/blob/5abff7a536ece42cbe53e96c4205b9625e923a49/src/frontend/src/spliter.rs#L19

For query requests, we should implement a `SystemTableScan` exec plan that forwards request to all datanodes.

## API change
Not applicable.

### LINKED ISSUE #1233: Support execute python script in distributed mode
### What type of enhancement is this?

API improvement

### What does the enhancement do?

Currently python script can only be executed in standalone mode. I think we can extend it to distributed mode. It should be fairly easy.

### Implementation challenges

_No response_

### PR DESCRIPTION
**Pull request by MichaelScofield**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Execute python script, both in standalone and distributed mode. Some refactor is part of #1010 , the `ScriptHandler` is implemented for Frontend instance instead of Datanode instance.  

The script table implementation in standalone mode is left untouched. In distributed mode, the script table is implemented as an ordinary `DistTable`, created under default catalog and schema.

The distributed script table has only one region, and place in only one datanode. Insertions and queries are just the same as other `DistTable`s. I think we can close #411 as well, distributed system tables can be all implemented this way.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)

close #1233
close #411

### CODE REVIEW

File: src/frontend/src/instance.rs [hunk 1]
```diff
@@ -132,23 +131,29 @@ impl Instance {
...
-            datanode_clients.clone(),
-        ));
+        let mut catalog_manager =
+            FrontendCatalogManager::new(meta_backend, partition_manager, datanode_clients.clone());
 
-        let dist_instance =
-            DistInstance::new(meta_client, catalog_manager.clone(), datanode_clients);
+        let dist_instance = DistInstance::new(
+            meta_client,
+            Arc::new(catalog_manager.clone()),
```
Comment (sunng87) [inline src/frontend/src/instance.rs:139]: Can we refactor this to avoid circular reference?

Comment (MichaelScofield) [reply src/frontend/src/instance.rs:139]: I've left a todo in the `dist_instance` field of FrontendCatalogManager. Once I've finished the work of creating distributed table by procedure, the circular dependency here will be eliminated.

**Review by sunng87 — APPROVED** [review]

**Comment by killme2008** [comment]
@MichaelScofield some conflicts have to be resolved.

File: src/frontend/src/catalog.rs [hunk 2]
```diff
@@ -106,9 +124,91 @@ impl CatalogManager for FrontendCatalogManager {
...
+            if self
+                .table(
+                    &request.catalog_name,
+                    &request.schema_name,
+                    &request.table_name,
+                )
+                .await?
+                .is_some()
+            {
+                return Ok(());
```
Comment (killme2008) [inline src/frontend/src/catalog.rs:142]: Maybe we should call open hook when table already exists?

File: src/frontend/src/catalog.rs [hunk 3]
```diff
@@ -106,9 +124,91 @@ impl CatalogManager for FrontendCatalogManager {
...
+            }
+
+            let time_index = request
```
Comment (fengjiachun) [inline src/frontend/src/catalog.rs:145]: What will happen if the following code is executed concurrently

Comment (MichaelScofield) [reply src/frontend/src/catalog.rs:147]: The safety of concurrently creating tables is ensured by "create table procedure" which I am currently doing.

**Comment by MichaelScofield** [comment]
resolved, @killme2008 PTAL

**Review by killme2008 — APPROVED** [review]
LGTM

**Pull request merged**